### PR TITLE
SPI peripheral API introduced for TizenRT

### PR DESCRIPTION
**Pull request by marcinsmsg**
1. New platform module for Tizen was created using CAPI System Peripheral API.
2. The iotjs.spec was extended with peripheral dependencies.
3. The operating system flag for Tizen build was changed from Linux to Tizen.
   We use Tizen specific APIs for hardware so this change is reasonable.
4. The compiler flag __linux__ was used in many places to check for Linux
   specific features. This flag is set in compiler not in IoT.js build
   environment. We need a way to distinguish between pure Linux (e.g. Raspbian)
   and Tizen, which is very specific flavour. It is better to move platform
   and API specific structures to platform files so ifdefs are not necessary.

IoT.js-DCO-1.0-Signed-off-by: Marcin m.ziombski@samsung.com

### CODE REVIEW

**Comment by marcinsmsg** [comment]
To pass travis tests, https://github.com/Samsung/TizenRT/pull/401 needs to be merged, as SPI library in TizenRT is missing a header providing size_t.

**Comment by yichoi** [comment]
Please make your code tidy https://travis-ci.org/Samsung/iotjs/jobs/264427795

**Comment by yichoi** [comment]
Is this for Tizen RT or Tizen ?

**Comment by yichoi** [comment]
Please rebase it to the master

**Comment by girishks2000** [comment]
please rename the patch title. Its TizenRT not Tizen

File: build.config [hunk 1]
```diff
@@ -111,7 +111,7 @@
       "nuttx": ["adc", "dgram", "gpio", "i2c", "pwm", "stm32f4dis", "uart"],
       "darwin": [],
       "tizen": ["adc", "ble", "dgram", "gpio", "i2c", "pwm", "spi", "uart", "https"],
-      "tizenrt": ["adc", "dgram", "gpio", "pwm", "uart"]
+      "tizenrt": ["adc", "dgram", "gpio", "pwm", "uart", "spi"]
     }
```
Comment (girishks2000) [inline build.config]: maintain alphabetical order of files. This is adhoc

**Comment by yichoi** [comment]
rebase please

File: src/platform/tizen/iotjs_module_spi-tizen.c [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+  }
+
+  return true;
+}
+
+
+bool iotjs_spi_close(iotjs_spi_t* spi) {
+  IOTJS_VALIDATED_STRUCT_METHOD(iotjs_spi_t, spi);
+
+  peripheral_spi_close(_this->platform->peripheral_spi);
```
Comment (hs0225) [inline src/platform/tizen/iotjs_module_spi-tizen.c]: Please check error. `peripheral_spi_close` returns a negative value if it fails.

File: src/platform/tizenrt/iotjs_module_spi-tizenrt.c [hunk 3]
```diff
@@ -108,13 +112,29 @@ void iotjs_spi_open_worker(uv_work_t* work_req) {
   SPI_WORKER_INIT;
   IOTJS_VALIDATED_STRUCT_METHOD(iotjs_spi_t, spi);
 
+  if (_this != 0) // use of _this to prevent compiler warnings
+  {
```
Comment (hs0225) [inline src/platform/tizenrt/iotjs_module_spi-tizenrt.c]:
```c
if () {
```

File: src/platform/tizenrt/iotjs_module_spi-tizenrt.c [hunk 4]
```diff
@@ -108,13 +112,29 @@ void iotjs_spi_open_worker(uv_work_t* work_req) {
...
+  }
+
   if (!iotjs_spi_open(spi)) {
-    DDLOG("%s - SPI open failed %d", __func__, _this->bus);
+    DDLOG("%s - SPI open failed %d", __func__, _this->platform->bus);
```
Comment (hs0225) [inline src/platform/tizenrt/iotjs_module_spi-tizenrt.c]: ditto.

File: src/platform/tizen/iotjs_module_spi-tizen.c [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+}
+
+
+void iotjs_spi_open_worker(uv_work_t* work_req) {
+  SPI_WORKER_INIT;
+  IOTJS_VALIDATED_STRUCT_METHOD(iotjs_spi_t, spi);
+
+  if (peripheral_spi_open(_this->platform->bus, _this->chip_select,
+                          &_this->platform->peripheral_spi) < 0) {
+    DDLOG("%s - SPI open failed %d", __func__, _this->platform->bus);
```
Comment (hs0225) [inline src/platform/tizen/iotjs_module_spi-tizen.c]: `DDLOG` is a warning message. You should use `DLOG` which is an error message.

**Comment by marcinsmsg** [comment]
I'm going on holidays so @pmarcinkiew is taking over this pull request.

**Pull request closed without merging**